### ISSUE TITLE
UWP AppShell Issues

### ISSUE THREAD

**Issue by rainman-63** [PrismLibrary/Prism#1628]
### Description
Couple of issues with creating UWP appshell apps.

1) UWP by default will inherit the parent's data context, so the OrchestrateAsync method will always find an "old" vm.  This is fairly simple to get around by executing the ViewModelLocator's SetAutowireViewModel in the pages constructor.

2) After working around the previous issue, we encountered another issue. When the page's constructor calls ViewModelLocator.SetAutowireViewModel, the UnityContainerExtension's method (ResolveViewModelForView) attempts to resolve the viewmodel, it appears to be assuming that since this is a page, then it must have a frame that will be used in navigation. 

I made this simple change to the ResolveViewModelForView method below to correct the issue.

```
public object ResolveViewModelForView(object view, Type viewModelType)
{
    if (view is Page page && page.Frame != null)
    {
        var service = NavigationService.Instances[page.Frame];
        ResolverOverride[] overrides = null;

        overrides = new ResolverOverride[]
        {
            new DependencyOverride(
                typeof(INavigationService),
                service
            )
        };
        return Instance.Resolve(viewModelType, overrides);
    }
    else
    {
        return Instance.Resolve(viewModelType);
    }
}

```

### Expected Behavior
Expect the page to be navigated to with the correct view model.

### Actual Behavior
When not using ViewModelLocator.SetAutowireViewModel, the shell's view model is inherited.

When using ViewModelLocator.SetAutowireViewModel, the constructor will throw an exception.

### Basic Information

- Prism Version:  7.2.0.679-ci
- UWP: 6.1.9 
- IDE: VS 2017 15.8.9

### Reproduction Link
Attached sample app which displays the issue.

[AppShellBug.zip](https://github.com/PrismLibrary/Prism/files/2595836/AppShellBug.zip)

**Comment by MikelThief** [comment]
For anybody who finds this issue on Google - it was fixed in version 7.2.0.690-ci.

**Comment by rainman-63** [comment]
Thanks @MikelThief
